Controller/Main.as

```diff
@@ -223,4 +246,33 @@ void SpecificPositionEntryCoroutine(ref@ time){
     if(positionEntry !is null && positionEntry.isValid()){
         leaderboardArrayTmp.InsertLast(positionEntry);
     }
-}
\ No newline at end of file
+}
+
+void UpdateTimeDifferenceEntry(array<LeaderboardEntry@> arrayTmp) {
+    if (currentComboChoice == -1) {
+        // timeDifferenceEntry is the entry that has entryType Pb
+        for (uint i = 0; i < arrayTmp.Length; i++) {
+            if (arrayTmp[i].entryType == EnumLeaderboardEntryType::PB) {
+                timeDifferenceEntry = arrayTmp[i];
+                break;
+            }
+        }
+    } else {
+        timeDifferenceEntry.time = -1;
+        timeDifferenceEntry.position = -1;
+        timeDifferenceEntry.entryType = EnumLeaderboardEntryType::POSITION; // Doesn't really matter since it isn't checked
+        for (uint i = 0; i < arrayTmp.Length; i++) {
+            if (arrayTmp[i].position == currentComboChoice && arrayTmp[i].entryType == EnumLeaderboardEntryType::POSITION) {
+                timeDifferenceEntry = arrayTmp[i];
+                break;
+            }
+        }
+    }
+}
+
+string TimeLogString(int time) {
```

Comment:
Could you move this to the toolbox please? feels like an util function to me :D

Reply:
Changes are done.